### PR TITLE
Correctly loop from preview point in song select

### LINKED ISSUE #3112: Preview points are not taken into account in some cases
If a beatmapset has multiple audio files and you switch between difficulties using different audio files in songselect while staying in that beatmapset, osu!lazer will not use the preview point to start the audio file for the newly selected difficulty, but rather start the audio from the beginning.
Video: https://streamable.com/4lzln

### LINKED ISSUE #4155: Preview points are not taken into account at all
In the beatmap selection screen, if you let the song end, osu!lazer will loop back to the start of the song instead of using the preview point.

This is similar to #3112, however this happens across all beatmapsets (even those with only one audio file used in all difficulties).

Is this an oversight or is this desired behavior?

osu!lazer version: [commit 718136a](https://github.com/ppy/osu/commit/718136a8928db9f1a3cd9122f33d6bb650820c90)

[Example](https://streamable.com/j91d8)

### PR DESCRIPTION
**Pull request by Aergwyn**
- [x] depends on https://github.com/ppy/osu-framework/pull/2118

Closes #4155
Closes #3112

### CODE REVIEW

File: osu.Game/Screens/Select/SongSelect.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
```
[the remark this answers is not in the document]
Comment (Aergwyn) [reply osu.Game/Screens/Select/SongSelect.cs]: I don't know. Happened on a file on the framework PR as well and I had to manually revert it. No clue how I should get rid of this here.

**Comment by Aergwyn** [comment]
Maybe the argument from `ensurePlayingSelected()` could be entirely removed? Is there a use-case where you don't want to use the preview point in any song select?

File: osu.Game/osu.Game.csproj [hunk 2]
```diff
@@ -18,7 +18,7 @@
     <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="2.2.1" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite.Core" Version="2.2.1" />
     <PackageReference Include="Newtonsoft.Json" Version="12.0.1" />
-    <PackageReference Include="ppy.osu.Framework" Version="2019.125.0" />
+    <PackageReference Include="ppy.osu.Framework" Version="0.0.8120" />
```
Comment (UselessToucan) [inline osu.Game/osu.Game.csproj]: You probably did not want to commit this change.

Comment (Aergwyn) [reply osu.Game/osu.Game.csproj]: That was actually intentional. Not sure how the procedure is but I wanted to keep this functional until the dependent framework PR is merged.

**Comment by peppy** [comment]
I've resolved the conflict / removed the line you had in player and can't notice any side effects. Could you please check what you were seeing going wrong without it?

Also, doesn't this feel a bit wrong? https://youtu.be/r8Rp7YLCvMU

**Comment by Aergwyn** [comment]
Mmmh this is hard to tell. But it feels like the sound is overlapping itself for a moment.

The one line you removed was one clock trying to catch up with another and creating a disturbing noise. I guess all the clock fixes and changes fixed that by itself now so it wasn't necessary anymore.

**Comment by peppy** [comment]
I mean the fact it is seeking at all in the recorded case feels a bit weird.

**Comment by Aergwyn** [comment]
It's the same as stable but what makes it feel weird, I guess, is that the fading is missing so it's a hard cut atm.

**Comment by peppy** [comment]
I think for the time being in this case we shouldn't restart. We can add it back once we have the ability to better transition the audio.

**Review by peppy — APPROVED** [review]

**Pull request merged**